dropwizard-db/src/main/java/io/dropwizard/db/AbstractDataSourceFactory.java

```diff
@@ -0,0 +1,66 @@
+package io.dropwizard.db;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.base.Optional;
+import io.dropwizard.util.Duration;
+
+import java.util.Map;
+
+/**
+ * Interface of an abstract JDBC connection pool
+ */
+public interface AbstractDataSourceFactory {
```

Comment:
In my opinion the prefix "Abstract" is not really a good choice for use in interfaces. Maybe "PooledDateSourceFactory" is more fitting in this case.